azurerm/internal/services/web/resource_arm_app_service_hybrid_connection.go

```diff
@@ -0,0 +1,186 @@
+package web
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/response"
+
+	"github.com/Azure/azure-sdk-for-go/services/web/mgmt/2018-02-01/web"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmAppServiceHybridConnection() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAppServiceHybridConnectionCreateUpdate,
+		Read:   resourceArmAppServiceHybridConnectionRead,
+		Update: resourceArmAppServiceHybridConnectionCreateUpdate,
+		Delete: resourceArmAppServiceHybridConnectionDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"app_service_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"resource_group_name": azure.SchemaResourceGroupName(),
+			"namespace_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(6, 50),
+			},
+			"relay_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"relay_arm_uri": {
```

Comment:
Given that this actually takes the ID would it make sense to rename it?
```suggestion
			"relay_id": {
```

and mention in the docs it maps to the arm uri property

Reply:
I was torn on this one, the API refers to it unhelpfully as relayArmURI, so anyone following the documentation would be (probably?) expecting this. My thought was to leave this and put appropriate comment in the doc page for it.

Reply:
We often slightly rename or correct the API to what a user would expect, `relay_id` is really clear to someone in autocomplete or a reading the tf, and makes  `relay_id = relat.test.id` make sense, while uri is only helpful to those coming from the API, in which case having that in the docs would assist them (and i think the latter case would be the les common)